File: java/src/me/kzv/datastructure/arrayList/ArrayListTest.java
package me.kzv.datastructure.arrayList;

import java.util.stream.IntStream;

public class ArrayListTest {
    public static void main(String[] args) {
        ArrayList<Integer> myArrayList = new ArrayList<>();

        IntStream.rangeClosed(1, 100).forEach(myArrayList::add);

        System.out.println(myArrayList.size());
        System.out.println(myArrayList.get(0));
        System.out.println(myArrayList.get(99));

        myArrayList.remove(0);
        System.out.println();

        System.out.println(myArrayList.size());
        System.out.println(myArrayList.get(0));
        System.out.println(myArrayList.get(99));
        System.out.println(myArrayList.get(98));
        System.out.println(myArrayList.get(97));
        int get96 = myArrayList.get(96);
        System.out.println(get96);

        boolean contains = myArrayList.contains(get96);
        System.out.println(contains);
        System.out.println(myArrayList.isEmpty());

        ArrayList<Integer>.Iterator it = myArrayList.iterator();
        while (it.hasNext()) {
            System.out.println(it.next());
        }
    }
}

File: kotlin/src/main/kotlin/me/kzv/datastructure/myarray/test/arrayListTest.kt
package me.kzv.datastructure.myarray.test

fun main(){
    intArrayList()
}

fun intArrayList(){
    val numbers = ArrayList<Int>()

    numbers.add(10)
    numbers.add(20)
    numbers.add(30)
    numbers.add(40)
    println("add(값)")
    println(numbers)

    //add(값)
    //[10, 20, 30, 40]
    //
    //add(인텍스, 값)
    //[10, 50, 20, 30, 40] <- 20을 한칸 밀어내고 그 사이에 값을 넣었다.

    numbers.add(1, 50)
    println("\nadd(인텍스, 값)")
    println(numbers)

    val it = numbers.iterator()
    while (it.hasNext()) {
        println(it.next())
    }

    for ((index, number) in numbers.withIndex()) {
        println("index: $index number: $number")
    }
    
}

